.github/workflows/full.yml

```diff
@@ -299,7 +299,7 @@ jobs:
         run: npm run lint
   crux_test:
     runs-on: ubuntu-22.04
-    needs: gather_changes
+    needs: [gather_changes, crux_lint]
```

Comment:
I'd run the test at the same time as lint/etc runs. Doesn't make much sense logically at first, (because why should we test if the formatting/linting arent good, right?) but it makes the pipeline faster a bit, and checking everything at the same time gives more broad feedback to the dev.